src/python/pants/build_graph/address.py

```diff
@@ -333,7 +340,11 @@ def spec(self) -> str:
         prefix = "//" if not self.spec_path else ""
         if self._relative_file_path is None:
             path = self.spec_path
-            target = "" if self._target_name is None and self.generated_name else self.target_name
+            target = (
+                ""
+                if self._target_name is None and (self.generated_name or self.parameters)
```

Comment:
This is a change. We used to do `a/b:b@k=v`, now it's `a/b@k=v`. I'm pretty confident that's what Stu had intended, only oversight.

If not, we can change back.